### PR TITLE
Adds picture frames.

### PR DESCRIPTION
**Pull request by harmonyn**
ass

Picture frames have been added to the game. Partially ported from Paradise Station, except with our photo sprites. One stack of wood makes a frame.

**WHY THIS ADDS TO THE GAME:** Allows more customization of areas. Hanging a photo in your office, creating an art gallery, etc. Gives assistants and clowns a way to non-harmfully prank. You get to FRAME YOUR FUCKING ASS.

:cl:
add: Nanotrasen supports the arts. We now offer picture frames!
/:cl:

### CODE REVIEW

File: code/modules/paperwork/photography.dm [hunk 1]
```diff
@@ -484,3 +485,126 @@
...
+	icon = 'icons/obj/decals.dmi'
+	icon_state = "frame-empty"
+	var/obj/item/weapon/photo/displayed
+
+/obj/item/weapon/picture_frame/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/weapon/photo))
+		if(!displayed)
+			var/obj/item/weapon/photo/P = I
+			displayed = P
+			displayed.icon = P.icon
```
Comment (AnturK) [inline code/modules/paperwork/photography.dm]: You set the icon to the same value here.

File: code/modules/paperwork/photography.dm [hunk 2]
```diff
@@ -484,3 +485,126 @@
...
+			qdel(P)
+			update_icon()
+		else
+			user << "<span class=notice>\The [src] already contains a photo.</span>"
+
+	..()
+
+/obj/item/weapon/picture_frame/attack_hand(mob/user)
+	if(user.get_inactive_held_item() != src)
```
Comment (AnturK) [inline code/modules/paperwork/photography.dm]: Really not a fan of this, pizza bombs was changed to this way of interaction and all it did was confuse people.

File: code/modules/paperwork/photography.dm [hunk 3]
```diff
@@ -484,3 +485,126 @@
...
+		..()
+		return
+	if(displayed)
+		user.put_in_hands(displayed)
+		user << "<span class='notice'>You carefully remove the photo from \the [src].</span>"
+		displayed = null
+		update_icon()
+
+/obj/item/weapon/picture_frame/attack_self(mob/user)
+	user.examinate(src)
```
Comment (AnturK) [inline code/modules/paperwork/photography.dm]: Especially when you basically throw away most common way of interaction.

File: code/modules/paperwork/photography.dm [hunk 4]
```diff
@@ -484,3 +485,126 @@
...
+/obj/item/weapon/picture_frame/examine(mob/user)
+	if(in_range(src, user) && displayed)
+		displayed.show(user)
+	else
+		..()
+
+/obj/item/weapon/picture_frame/update_icon()
+	overlays.Cut()
+	if(displayed)
+		overlays |= getFlatIcon(displayed)
```
Comment (AnturK) [inline code/modules/paperwork/photography.dm]: You can just do overlays = list() and save a bit on time.

File: code/modules/paperwork/photography.dm [hunk 5]
```diff
@@ -484,3 +485,126 @@
...
+	else
+		icon_state = initial(icon_state)
+
+/obj/item/weapon/picture_frame/afterattack(atom/target, mob/user, proximity)
+	if(isturf(target) && proximity)
```
Comment (AnturK) [inline code/modules/paperwork/photography.dm]: You can just do iswallturf directly here.

File: code/modules/paperwork/photography.dm [hunk 6]
```diff
@@ -484,3 +485,126 @@
...
+		if(!do_after(user, 30*O.toolspeed, target = src))
+			return
+		playsound(src, 'sound/items/Deconstruct.ogg', 50, 1)
+		user.visible_message("<span class='notice'>[user] unfastens [src].</span>", \
+							 "<span class='notice'>You unfasten [src].</span>")
+		var/obj/item/weapon/picture_frame/F = new /obj/item/weapon/picture_frame(get_turf(user))
+		if(framed)
+			F.displayed = framed
+			F.displayed.loc = framed.loc
+			F.displayed.icon = framed.icon
```
Comment (AnturK) [inline code/modules/paperwork/photography.dm]: All these icon transcriptions are not needed, also set framed to null

**Review by AnturK — COMMENTED** [review]
Not sure what would cause the icon loss, but there's a lot of unneeded code here.

File: code/modules/paperwork/photography.dm (hunk 1 above)
Comment (MrPerson) [inline code/modules/paperwork/photography.dm]: This line seems redundant

File: code/modules/paperwork/photography.dm [hunk 7]
```diff
@@ -484,3 +485,126 @@
...
+		if (iswallturf(T))
+			user.visible_message("<span class='notice'>[user] fastens \the [src] to [T].</span>", \
+								 "<span class='notice'>You fasten \the [src] to [T].</span>")
+			var/obj/structure/sign/picture_frame/newpic = new /obj/structure/sign/picture_frame(T)
+			newpic.icon_state = icon_state
+			newpic.overlays = overlays.Copy()
+			if(displayed)
+				newpic.framed = displayed
+				newpic.framed.icon = displayed.icon
+				newpic.framed.loc = src
```
Comment (MrPerson) [inline code/modules/paperwork/photography.dm]: This should be newpic.framed.forceMove(src)

File: code/modules/paperwork/photography.dm [hunk 8]
```diff
@@ -484,3 +485,126 @@
 	C.toner -= 20	 //Cyborgs are very ineffeicient at printing an image
 	visible_message("[C.name] spits out a photograph from a narrow slot on its chassis.")
 	usr << "<span class='notice'>You print a photograph.</span>"
+
+// Picture frames
+
+/obj/item/weapon/picture_frame
```
Comment (Core0verload) [inline code/modules/paperwork/photography.dm]: \>not using wall frame systems
Whyyyyy?

**Comment by coiax** [comment]
Replace the bartender's picture of Pun Pun with this system, for #consistency

File: code/modules/paperwork/photography.dm (hunk 1 above)
Comment (bgobandit) [inline code/modules/paperwork/photography.dm]: I told you I was having issues with there being no icon when removing photos, this doesn't work but I don't know how to make it work

File: code/modules/paperwork/photography.dm (hunk 2 above)
Comment (bgobandit) [inline code/modules/paperwork/photography.dm]: stacks use this kind of interaction too, so do a lot of things

File: code/modules/paperwork/photography.dm (hunk 3 above)
Comment (bgobandit) [inline code/modules/paperwork/photography.dm]: consistent with photos though

File: code/modules/paperwork/photography.dm (hunk 8 above)
Comment (bgobandit) [inline code/modules/paperwork/photography.dm]: I could use that yeah

Comment (bgobandit) [inline code/modules/paperwork/photography.dm]: update I have tried using the wall frame system and it fucked everything up, the functionality is just way too different

**Comment by dannno** [comment]
cool idea

**Comment by bgobandit** [comment]
I can do that, however, the Pun Pun photo would have to be a static sprite (like asses are) and not a typepath photo, because that way lies hell (photo code is hell)

**Comment by AnturK** [comment]
Oh, something i forgot to mention, right now photos scale with that white border in mind, maybe it's better idea for photos to store the full scale photo too, then these will look better.

**Comment by bgobandit** [comment]
Possibly, yeah. I think the """"lore"""" is that cameras produce Polaroids, which would be framed. But it would require redoing the sprite and that took a good hour

**Comment by optimumtact** [comment]
I removed the optional name here from your changelog as it will use your github name by default, otherwise you would show up in teh changelog as "optional name here"

**Pull request merged**